src/Core/PolynomialFunction.java

```diff
@@ -1,17 +1,20 @@
 package Core;
 
 import java.awt.*;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
 import java.util.List;
+import java.util.concurrent.*;
 import java.util.regex.*;
-import java.util.Arrays;
 import java.util.*;
 
 public class PolynomialFunction implements ParametricFunction {
     private String functionString;
     private double[] coefficients;
-    public List<Double> roots;
-    public List<Double> extremePoints;
+    public Set<Double> roots;
+    public Set<Double> extremePoints;
     public Color graphColor;
+    private final ExecutorService executorService = Executors.newFixedThreadPool(4);
```

Comment:
from what I saw, we only ever use 2 threads simultaneously. Why is this initialized to 4 threads?

Reply:
Oh thats because i was stupid and understood it wrongly at the beginning and just thought more threads more power. I later checked how it worked but forgot to change it back. Thanks for noting ^^

Reply:
Done